src/lib/u2/browser.js

```diff
@@ -0,0 +1,574 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'ViewFactory',
+  extends: 'foam.core.Function',
+  properties: [
+    {
+      name: 'adapt',
+      value: function(old, nu) {
+        if ( typeof nu === 'function' ) return nu;
```

Comment:
I wonder if this should be done via property flavours instead of adapt.

{
  class: 'ViewFactory',
  name: 'rowView',
}

this.rowView$f(args)

It would let you use value: 'someView' rather than factory: function() { return 'someView'; }, like you did before, and could help preserve serializability.